.azurepipelines/templates/pr-gate-build-job.yml

```diff
@@ -63,6 +63,9 @@ jobs:
   pool:
     vmImage: ${{ parameters.vm_image }}
 
+  ${{ if contains(parameters.tool_chain_tag, 'GCC') }}:
+    container: 'ghcr.io/tianocore/containers/fedora-35-test:46de622'

```

Comment:
(will duplicate these comments in mailing list)

Consider using the container as the input rather than assuming based on toolchain. This would give us more flexibility for different build images and tool chains. This is what we experimented with in MU. https://github.com/microsoft/mu_basecore/pull/108/files#diff-bc122674ce9d34eacd6719c68b21f00eae777b4a2f96cb593b91844fbae3bd0b